server/util/repeatingRide.ts

```diff
@@ -0,0 +1,115 @@
+  schedule.scheduleJob('0 5 10 * * *', () => {
+    createRepeatingRides();
+  });
+}
+
+
+function createRepeatingRides() {
+  let tomorrowDate = moment(new Date());
+  tomorrowDate = tomorrowDate.add(1, 'days');
+  const tomorrowDateString = tomorrowDate.format('MM/DD/YYYY');
+
+  const tomorrowDay = tomorrowDate.weekday();
+
+  // condition: recurring ride that includes tomorrowDay
+  const condition = new Condition()
+    .where('recurring')
+    .eq(true)
+    .and()
+    .where('recurringDays')
+    .contains(tomorrowDay);
+
+  Ride.scan(condition).exec((_, data) => {
+    data?.forEach(async (masterRide) => {
+      const {
+        id, rider, driver, startLocation, endLocation, startTime, endTime, endDate, edits,
+      } = masterRide.toJSON() as RideType;
+
+      // only continue if the endDate has not passed
+      if (endDate && endDate >= tomorrowDateString) {
+        const tomorrowDateOnly = moment(new Date()).add(1, 'days').format('YYYY-MM-DD');
+
+        // the repeating ride's instance start and end times use tomorrow's date
+        const newStartTimeOnly = moment(startTime).format('HH:mm:ss');
+        const newStartTime = moment(`${tomorrowDateOnly}T${newStartTimeOnly}`).toISOString();
+
+        const newEndTimeOnly = moment(endTime).format('HH:mm:ss');
+        const newEndTime = moment(`${tomorrowDateOnly}T${newEndTimeOnly}`).toISOString();
+
+        const repeatingRide = new Ride({
+          id: uuid(),
+          rider,
+          startLocation,
+          endLocation,
+          startTime: newStartTime,
+          requestedEndTime: newEndTime,
+          endTime: newEndTime,
+          driver,
+          recurring: false,
+        });
+
+        if (edits?.length) {
+          handleEdits(edits, tomorrowDateOnly, repeatingRide, masterRide);
+        } else {
+          repeatingRide.save().catch((err) => console.log(err));
+        }
+      }
+    });
+  });
+}
+
+function handleEdits(
+  edits: string[],
+  tomorrowDateOnly: string,
+  repeatingRide: AnyDocument,
+  masterRide: AnyDocument,
+) {
+  const seenEdits: string[] = [];
+  let editCount = 0;
+  const numEdits = edits.length;
+
+  edits.forEach(async (editId) => {
+    Ride.get(editId).then(async (editRide) => {
+      const editDate = moment(editRide.startTime);
+      const editDateOnly = moment(editDate).format('YYYY-MM-DD');
+
+      // only look at edits that match tomorrow's date
+      if (editDateOnly === tomorrowDateOnly) {
+        seenEdits.push(editId);
+        // if deleted = true
+        if (editRide.deleted) {
+          // create repeating ride instance only if it does NOT match the delete edit
+          if (!(repeatingRide.startTime === editRide.startTime
+            && repeatingRide.endTime === editRide.endTime
+            && repeatingRide.rider.id === editRide.rider.id
+            && repeatingRide.startLocation.id === editRide.startLocation.id
+            && repeatingRide.endLocation.id === editRide.endLocation.id)) {
```

Comment:
when would a repeating ride not match the delete edit?

Reply:
that would be if an instance is edited and then the master ride is later edited. Then the delete edit won't match the master ride, and I think from the frontend the fake instance would show up for that day, so they would have to delete that instance again for it to be removed

Reply:
just talked with nina, if the master ride is changed after an edit is made, the edit should be preserved and a new repeating ride shouldnt be created